blueocean-pipeline-api-impl/src/main/java/io/jenkins/blueocean/rest/impl/pipeline/PipelineNodeGraphVisitor.java

```diff
@@ -231,18 +231,23 @@ protected void handleChunkDone(@Nonnull MemoryFlowChunk chunk) {
         }else if (firstExecuted == null) {
             status = new NodeRunStatus(GenericStatus.NOT_EXECUTED);
         }else if(chunk.getLastNode() != null){
-            status = new NodeRunStatus(StatusAndTiming
-                                           .computeChunkStatus2(run, chunk.getNodeBefore(),
-                                                                firstExecuted, chunk.getLastNode(), chunk.getNodeAfter()));
+            // StatusAndTiming.computeChunkStatus2 seems to return wrong status for parallel sequential stages
+            // so check check if active and in the case of nested sequential
+            if(parallelNestedStages && chunk.getFirstNode().isActive()) {
+                status = new NodeRunStatus(chunk.getFirstNode());
```

Comment:
Does this only fix the issue if the `sleep` is in the first sequential stage? What if the test case is the following:

    pipeline {
        agent any
        stages {
            stage('Parallel') {
                parallel {
                    stage('Nested A') {
                        steps {
                            echo 'A'
                        }
                    }
                    stage('Nested B') {
                        stages {
                            stage('Nested B-1') {
                                steps {
                                    echo 'B'
                                }
                            }
                            stage('Nested B-2') {
                                steps {
                                    sleep time: 10, unit: 'SECONDS'
                                }
                            }
                        }
                    }
                }
            }
        }
    }

Reply:
yes no problem that work.
I think you are a bit of confused by the method name (getFirstNode) there is nothing related to the number of stages. A chunk is a kind of container with multiple nodes on it.

Reply:
@olamy Yeah I was definitely confused by the name, thanks for answering!